Tests/MongoSwiftSyncTests/SyncMongoClientTests.swift

```diff
@@ -320,4 +320,30 @@ final class SyncMongoClientTests: MongoSwiftTestCase {
         expect(try MongoClient("mongodb://localhost:12345", options: options))
             .to(throwError(errorType: MongoError.InvalidArgumentError.self))
     }
+
+    func testConnectionTimeout() throws {
+        let setupClient = try MongoClient.makeTestClient()
+        guard try setupClient.supportsFailCommand() else {
+            printSkipMessage(testName: self.name, reason: "failCommand not supported")
+            return
+        }
+
+        var failOptions = MongoClientOptions()
+        failOptions.connectTimeoutMS = 100
+        failOptions.serverSelectionTimeoutMS = 1000
+        let timeoutClient = try MongoClient.makeTestClient(options: failOptions)
+
+        let fp = FailPoint.failCommand(failCommands: ["isMaster"], mode: .alwaysOn, blockTimeMS: 500)
+        try fp.enable()
+        defer { fp.disable() }
+
+        expect(try timeoutClient.db("admin").runCommand(["ping": 1])).to(throwError())
```

Comment:
there looks to be an issue on 4.0 on Evergreen. it looks like blockConnection actually isn't supported on server versions < 4.2, so we should skip this test on those:
https://jira.mongodb.org/browse/SERVER-41070

Reply:
Specifically, it looks like 4.2.9 was the first server version to support it.